src/clj_uuid.clj

```diff
@@ -666,11 +745,79 @@
       (as-byte-array local-part))))
 
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Lexically Sortable UUID                       [RFC9562:5.7: UUID Version 7];;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn v7
+  "Generate a v7 unix time-based, lexically sortable UUID with monotonic
+  counter and cryptographically secure random portion. The 12 bit
+  rand_a is used as counter, rand_b is CSPRNG. Random numbers are
+  generated using the JVM default implementation of
+  java.security.SecureRandom."
+  ^java.util.UUID
+  []
+  (let [[t counter]     (clock/monotonic-unix-time-and-random-counter)
+        time            (bitmop/ldb #=(bitmop/mask 48  0) t)
+        ver-and-counter (bitmop/dpb #=(bitmop/mask 4  12) counter 0x7)
+        msb             (bit-or ver-and-counter (bit-shift-left time 16))
+        lsb             (bitmop/dpb #=(bitmop/mask 2 62) (random/long) 0x2)]
+    (UUID. msb lsb)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Custom UUID                                   [RFC9562:5.8: UUID Version 8];;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn v8
+  "Generate a v8 custom UUID with up to 122 bits of user data."
+  ^java.util.UUID
+  [^long msb ^long lsb]
+  (UUID.
+   (bitmop/dpb #=(bitmop/mask 4 12) msb 0x8)
+   (bitmop/dpb #=(bitmop/mask 2 62) lsb 0x2)))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; Predicates
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+(defn- compare-many
+  [f x y more]
+  (if (f x y)
+    (if (next more)
+      (recur f y (first more) (next more))
+      (f y (first more)))
+    false))
+
+(defn =
```

Comment:
oh, right, variadic protocol arguments.  I see your point.  Well, let me see if i can come up with alternative.  otherwise, not the end of the world unless anyone :use s the namespace or :refer :all